tests/test_other.py

```diff
@@ -19,13 +19,12 @@ def test_nonexistent_path(self):
         for key in list(sys.modules.keys()):
             if "snopt" in key:
                 sys.modules.pop(key)
-        with self.assertWarns(UserWarning):
-            module = try_import_compiled_module_from_path("snopt", "/a/nonexistent/path", raise_warning=True)
-            self.assertTrue(isinstance(module, str))
+        with self.assertRaises(ImportError):
```

Comment:
Can you add a test on `on_error='return'`?

Reply:
Done